src/NuGet.Core/NuGet.Commands/RestoreCommand.cs

```diff
@@ -240,9 +240,22 @@ public Task<RestoreResult> ExecuteAsync()
             context.LocalLibraryProviders.Add(
                 new SourceRepositoryDependencyProvider(nugetRepository, _logger, _request.CacheContext));
 
-            foreach (var provider in _request.Sources.Select(s => CreateProviderFromSource(s, _request.CacheContext)))
+            if (_request.SharedCache.RemoteProviders.Any())
```

Comment:
Same question as above. Why do we have two code paths here. Can't we consolidate and demand for them to be passed in?